### PR TITLE
[CLUSTER] Node state and API collector runner selector

### PR DESCRIPTION
**Pull request by ELDiablO59152**
## REQUIREMENTS
- https://github.com/VultureProject/vulture-base/pull/99
### Added
- [CLUSTER] Node state and heartbeat
- [GUI] API collector node runner selector

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -21,6 +21,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - [API_PARSER] NEW collector **CSC_DOMAINMANAGER**
 - [LOG_FORWARDER] Disk-assisted queues for reliability
 - [PORTAL] [IDP] Refresh token support through OAuth2 flow
+- [CLUSTER] Node state and heartbeat
+- [GUI] API collector node runner selector
```
Comment (frikilax) [inline CHANGELOG:25]: must be moved to [Unreleased] version

File: vulture_os/gui/crontab/api_clients_parser.py [hunk 2]
```diff
@@ -84,9 +92,13 @@ def api_clients_parser():
...
-        p = Process(target=execute_parser, name=frontend.name, args=(frontend.to_dict(),))
-        p.start()
-        processes.append(p)
+        if node_selected(current_node, frontend):
+            p = Process(target=execute_parser, name=frontend.name, args=(frontend.to_dict(),))
+            p.start()
+            processes.append(p)
+
+    if not processes:
+        return
```
Comment (frikilax) [inline vulture_os/gui/crontab/api_clients_parser.py:101]: seems superfluous, the case is handled properly in the while loop

File: vulture_os/gui/crontab/api_clients_parser.py [hunk 3]
```diff
@@ -70,11 +70,19 @@ def execute_parser(frontend):
...
+def node_selected(current_node, frontend):
+    if current_node.state == "UP":
+        if frontend.node == current_node:
+            return True
+        elif current_node.is_master_mongo:
+            if not frontend.node:
+                return True
+            elif frontend.node.state != "UP":
+                return True
+    return False
```
Comment (frikilax) [inline vulture_os/gui/crontab/api_clients_parser.py:82]: This function needs comments to clarify what conditions are tested

File: vulture_os/gui/management/commands/toggle_maintenance.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+            raise CommandError('Error: Node is not bootstrapped')
+
+        if node := Cluster.get_current_node():
+            try:
+                if node.state == "MAINTENANCE":
+                    node.set_state("DOWN")
+                else:
+                    node.set_state("MAINTENANCE")
+            except Exception as e:
+                self.stdout.write(self.style.SUCCESS(f"{e}\n\nPlease relaunch this script after solving the issue."))
```
Comment (frikilax) [inline vulture_os/gui/management/commands/toggle_maintenance.py:22]:
```suggestion
                if node.state == "MAINTENANCE":
                    node.set_state("DOWN")
                    self.stdout.write(self.style.SUCCESS(f"The Node has been put out of MAINTENANCE."))
                else:
                    node.set_state("MAINTENANCE")
                    self.stdout.write(self.style.SUCCESS(f"The Node is now in MAINTENANCE."))
            except Exception as e:
                self.stdout.write(self.style.ERROR(f"{e}\n\nPlease relaunch this script after solving the issue."))
```

File: vulture_os/gui/views/install.py
Comment (frikilax) [inline vulture_os/gui/views/install.py:1]: Those modifications have nothing to do with the Feature, please include them in a separate PR

File: vulture_os/services/frontend/form.py
Comment (frikilax) [inline vulture_os/services/frontend/form.py:1]: Those modifications have nothing to do with the Feature, please include them in a separate PR

File: vulture_os/services/frontend/models.py
Comment (frikilax) [inline vulture_os/services/frontend/models.py:1]: This modification have nothing to do with the Feature, please include it in a separate PR

File: vulture_os/system/templates/system/cluster.html [hunk 5]
```diff
@@ -73,7 +73,37 @@ <h4 class="modal-title">{% trans "Cluster configuration" %}</h4>
...
+              break;
+            case "UNKNOWN":
+              result += '<i class="icon fa fa-question-circle"></i>&nbsp;';
+              break;
+            case "DISABLED":
+              result += '<i class="icon fa fa-toggle-off"></i>&nbsp;';
+              break;
+            case "WAITING":
+              result += '<i class="icon fa fa-spinner fa-spin"></i>&nbsp;';
+              break;
```
Comment (frikilax) [inline vulture_os/system/templates/system/cluster.html:101]:
```suggestion
```

**Review by frikilax — CHANGES_REQUESTED** [review]

File: CHANGELOG (hunk 1 above)
Comment (ELDiablO59152) [reply CHANGELOG]: Done

File: vulture_os/gui/crontab/api_clients_parser.py (hunk 2 above)
Comment (ELDiablO59152) [reply vulture_os/gui/crontab/api_clients_parser.py]: Deleted as suggested

File: vulture_os/gui/crontab/api_clients_parser.py (hunk 3 above)
Comment (ELDiablO59152) [reply vulture_os/gui/crontab/api_clients_parser.py]: Comments added

File: vulture_os/gui/management/commands/toggle_maintenance.py (hunk 4 above)
Comment (ELDiablO59152) [reply vulture_os/gui/management/commands/toggle_maintenance.py]: Good poinr, output logs added

File: vulture_os/system/templates/system/cluster.html (hunk 5 above)
Comment (ELDiablO59152) [reply vulture_os/system/templates/system/cluster.html]: Suggestion applied, this was unused

**Review by ELDiablO59152 — COMMENTED** [review]
Ready to merge

**Pull request merged**